core/tests/gateway-migration-test/tests/migration.test.ts

```diff
@@ -159,16 +163,31 @@ describe('Migration From/To gateway test', function () {
             tryCount += 1;
             await utils.sleep(1);
         }
+    });
 
-        // Additionally wait until the priority queue is empty if we are migrating to gateway
-        if (direction == 'TO') {
-            let priorityQueueSize = await l1MainContract.getPriorityQueueSize();
-            let tryCount = 0;
-            while (priorityQueueSize > 0 && tryCount < 30) {
-                priorityQueueSize = await l1MainContract.getPriorityQueueSize();
-                tryCount += 1;
-                await utils.sleep(1);
+    step('Pause deposits before initiating migration', async () => {
+        await utils.spawn(`zkstack chain pause-deposits --chain ${fileConfig.chain}`);
+
+        // We need to wait for a new block for the deposits paused time check to pass
+        const startBlock = await tester.ethProvider.getBlock('latest');
+        let tryCount = 0;
+        while ((await tester.ethProvider.getBlock('latest'))?.timestamp! <= startBlock?.timestamp! && tryCount < 30) {
+            tryCount += 1;
+            await utils.sleep(1);
+        }
+
+        // Additionally wait until the priority queue is empty
+        let getPriorityQueueSize = async () => {
+            if (direction == 'TO') {
+                return await l1MainContract.getPriorityQueueSize();
+            } else {
+                return await gwMainContract.getPriorityQueueSize();
```

Comment:
why are we checking the GW PQ size? What if there are alternative chains, they can keep adding txs there, no? 

Maybe the only solution is to query the chains DP both on L1 and on GW